packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -63,7 +100,36 @@ abstract class FlutterCommand extends Command {
 
   Future<int> runInProject();
 
+  List<Device> get devicesForCommand => _devicesForCommand;
+
+  Device get deviceForCommand {
+    // TODO(devoncarew): Switch this to just supporting one connected device?
+    return devicesForCommand.isNotEmpty ? devicesForCommand.first : null;
+  }
+
+  // This is caculated in run() if the command has [requiresDevice] specified.
+  List<Device> _devicesForCommand;
+
   ApplicationPackageStore applicationPackages;
   Toolchain toolchain;
   DeviceStore devices;
+
+  Future<List<Device>> _calculateDevicesForCommand() async {
```

Comment:
I also like to use "compute" prefix to emphasize that a non-trivial amount of work is done. "find" implies that too a little. "get" and "lookup" may not be very clear about this, especially "get".